### PR TITLE
Netavark interface

### PR DESCRIPTION
**Pull request by Luap99**
#### Make networking code reusable 

To prevent code duplication when creating new network backends move
reusable code into a separate internal package.

This allows all network backends to use the same code as long as they
implement the new NetUtil interface.

#### netavark network interface

Implement a new network interface for netavark.
For now only bridge networking is supported.
The interface can create/list/inspect/remove networks. For setup and
teardown netavark will be invoked.

#### netavark IPAM assignment

Add a new boltdb to handle IPAM assignment.

The db structure is the following:
Each network has their own bucket with the network name as bucket key.
Inside the network bucket there is an ID bucket which maps the container ID (key)
to a json array of ip addresses (value).
The network bucket also has a bucket for each subnet, the subnet is used as key.
Inside the subnet bucket an ip is used as key and the container ID as value.

The db should be stored on a tmpfs to ensure we always have a clean
state after a reboot.

### CODE REVIEW

**Comment by Luap99** [comment]
@baude @mheon PTAL I think this should be ready. The default network backend is still cni. You can change it via config or `--network-backend` flag to netavark. You can check the used backend with `podman info --format {{.Host.NetworkBackend}}`
The netavark tests that need to execute netavark will be skipped.

File: libpod/network/internal/util/bridge.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/pkg/errors"
+)
+
+func CreateBridge(n NetUtil, network *types.Network, usedNetworks []*net.IPNet) error {
+	if network.NetworkInterface != "" {
+		bridges := GetBridgeInterfaceNames(n)
+		if pkgutil.StringInSlice(network.NetworkInterface, bridges) {
+			return errors.Errorf("bridge name %s already in use", network.NetworkInterface)
+		}
+		if !define.NameRegex.MatchString(network.NetworkInterface) {
```
Comment (rhatdan) [inline libpod/network/internal/util/bridge.go:19]: Nit: Should this be checked first before loading the bridges?

File: libpod/network/internal/util/create.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		network.Labels = map[string]string{}
+	}
+	if network.Options == nil {
+		network.Options = map[string]string{}
+	}
+	if network.IPAMOptions == nil {
+		network.IPAMOptions = map[string]string{}
+	}
+
+	var name string
```
Comment (rhatdan) [inline libpod/network/internal/util/create.go:20]: Why are these predefined, should just use them locally.

**Review by rhatdan — APPROVED** [review]

**Pull request merged**